### PR TITLE
[all] upgrade openshift-client to v4.2.2

### PR DESCRIPTION
**Pull request by dsimansk**
Another take on client upgrade.

I wonder if we should implement some tests based on [mock servers](https://github.com/fabric8io/kubernetes-client/tree/master/openshift-server-mock/src/main/java/io/fabric8/openshift/client/server/mock) to be on a safe side. E.g. I haven't spotted any `Endpoints` tests even though the API was broken with double-plural issue in `v4.1.3`.

### CODE REVIEW

**Comment by Crumby** [comment]
@dsimansk I'm not sure if mock server could discover this issue but adding tests can't  cause any harm, can it? Did you try to run some tests against OpenShift with the update?

**Comment by dsimansk** [comment]
@Crumby we actually I'm trying to solve one listing issue right now. Pls do not merge yet. I'll let you know the result.

**Comment by Crumby** [comment]
@dsimansk Is this still WIP?

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java [hunk 1]
```diff
@@ -644,96 +643,81 @@ public RoleBinding getRoleBinding(String name) {
...
 		RoleBinding roleBinding = getRoleBinding(name);
 
 		if (roleBinding == null) {
+			// e.g `view` is ClusterRole in K8s RBAC
+			// if the roles isn't present fallback to ClusterRole
+			String kind = rbac().roles().withName(name).get() == null ? "ClusterRole" : "Role";
 			roleBinding = new RoleBindingBuilder()
 					.withNewMetadata().withName(name).endMetadata()
-					.withNewRoleRef().withName(name).endRoleRef()
+					.withNewRoleRef().withKind(kind).withName(name).endRoleRef()
```
Comment (dsimansk) [inline core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: @Crumby imo this's ready. However, I'm not particularly sure about this "opinionated" part of code. I seemed to work in my tests, but...

**Comment by maschmid** [comment]
I think we should do a release before merging this one.

File: pom.xml [hunk 2]
```diff
@@ -44,7 +44,7 @@
 
 		<!-- Depedency version properties -->
 		<version.httpclient>4.5.5</version.httpclient>
-		<version.openshift-client>4.1.1</version.openshift-client>
+		<version.openshift-client>4.2.0</version.openshift-client>
```
Comment (simkam) [inline pom.xml]: 4.2.2 is already out, would be nice to upgrade directly to that version

Comment (dsimansk) [reply pom.xml]: Sure, that shouldn't be a problem.

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java (hunk 1 above)
Comment (maschmid) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: maybe make it explicit (have  addClusterRoleToServiceAccount  )

Comment (dsimansk) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java]: I do agree that'd make the methods cleaner. However, if we split the API to `addRole...` and `addClusterRole...`, it'll eventually break all the usage where `view` is added to SA.

**Comment by dsimansk** [comment]
@Crumby @maschmid 
After further discussion with Marek. I done the following changes.
- Added `addRole*` methods that accept new parameter `String roleKind`
- Changed `getOrCreateRoleBinding` to substitute roleKind for `{admin, edit, view}` that are considered [user-facing roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles)
- Deprecated method `addRoleToGroup` to redirect users to new method with roleKind param
- Didn't add handling of `ClusterRoles` those can be still accessed via Fabric8's API

**Pull request merged**